content/algorithms/lca/LCA.md

```diff
@@ -0,0 +1,224 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.13.8
+kernelspec:
+  display_name: Python 3
+  name: python3
+---
+
++++ {"id": "jrizb5yufXBG"}
+
+# Lowest Common Ancestor
+
+In this tutorial, we will explore the lowest common ancestor algorithm implemented in networkx under `networkx/algorithms/lowest_common_ancestor.py`.
+
+## Definitions
+
+Before diving into the algorithm, let's first remember the concepts of an ancestor node and a descendant node.
+
+- **Ancestor:**
+  Given a rooted tree, any node $u$ which is on the path from root node to $v$ is an ancestor of $u$.
+
+- **Descendant:** A descendant of a node is either a child of the node or a child of some descendant of the node.
+- **Lowest Common Ancestor:** For two of nodes $u$ and $v$ in a tree, the lowest common ancestor is the lowest (i.e. deepest) node which is an ancestor of both $u$ and $v$.
+
++++ {"id": "Z5VJ4S_mlMiI"}
+
+## Example
+
+It is always a good idea to learn concepts on an example. Consider the following evalutionary tree. We will draw directed version of it and define the ancestor/descendant relationships.
+
+![image:evolutionary tree](images/evolutionary_tree.png)
+
++++ {"id": "-mvVopP42kk9"}
+
+Let's first draw the tree using NetworkX.
+
+```{code-cell}
+import matplotlib.pyplot as plt
+import networkx as nx
+from networkx.drawing.nx_pydot import graphviz_layout
```

Comment:
Just a quick note: Could you use `from networkx.drawing.nx_agraph import graphviz_layout` instead of `nx_pydot`? We would like to deprecate `pydot` :)